### PR TITLE
Portal Restructure

### LINKED ISSUE #6523: Portal remove custom theme
### Describe the bug
Step one to remove original custom theme based on Bootstrap 2 and essentially refactored out over the years.
Will remove from all header calls and replace with selectable light or dark openemr theme which for now will be selected from portal Config options.
I'll add to portal patient selectable once I get around to adding a patient account setup profile page so patient has better control over what is important to them.

I'm heading towards a new configurable landing dashboard to allow including custom modules/portal apps/plugins. Stephen is close to our first SMART app for portal and we will be adding some portal api endpoints. Good stuff.

Essentially what I'm after is to open the portal up for app developers and better vendor support which hopefully will incentivize folks to contribute back.

We need notifications for such things as when clinician sends secure message to patient the patient will get an email and/or SMS alert. Or send SMS/Email from dashboard to alert for patient to complete pending forms, launch a questionnaire or send onetime token for a specific task.

Also as we are making portal more app centric, included native components within portal tree should stay framework free and those apps wanting to use frameworks such as angular/react etc should be as custom modules. This will make it easier to manage the various dependencies I see flooding in.
 Unsure what I'll do about the AngularJS apps(secure messaging, secure chat) since that is a dead framework.

### Client configuration

- Browser: All
- OpenEMR version: v7.0.1
- Operating system: All

### PR DESCRIPTION
**Pull request by sjpadgett**
I'll cleanup build once I go through and pull any custom classes I may need.

Fixes #6523 

#### Short description of what this resolves:
Refactor portal to use core openemr theme builds. Might speed up builds some.
See issue and forum thread

#### Changes proposed in this pull request:
* initial custom css removal from headers
* Globals support portal themes
* add config portal theme options
* config yaml support
* Header class support

### CODE REVIEW

**Comment by sjpadgett** [comment]
Will you guys check me on Header class, globals and yaml changes. I really like that Header class btw..
Anyone wishing to practice up on BS4 feel free as I'm paying $1.99 / hr.:)

I would love to make portal BS5. Too much overlap now since recent move of some scripts to under core

File: config/config.yaml [hunk 1]
```diff
@@ -66,24 +66,17 @@ assets:
...
-        basePath: '%webroot%/public/themes/'
-        link: patientportal-base.css
-        rtl:
-            basePath: '%webroot%/public/themes/'
-            link: rtl_patientportal-base.css
+    portal-theme:
+      alreadyBuilt: true
+      link: '%portal_css_header%'
+      autoload: false
+      allowNoLoad: true
```
Comment (bradymiller) [inline config/config.yaml:73]: can remove above 2 lines (autoload defaults to false and allowNoLoad is only pertinent when autoload is true)

File: library/globals.inc.php [hunk 2]
```diff
@@ -3057,6 +3057,16 @@ function gblTimeZones()
...
 
+        'portal_css_header' => array(
+            xl('Portal Default Theme'),
+            array(
+                'style_light.css' => xl('Light'),
+                'style_dark.css' => xl('Dark')
+            ),
+            'style_light.css',
+            xl('Pick a default portal theme.')
+        ),
```
Comment (bradymiller) [inline library/globals.inc.php:3068]: Would be cool if also made this a patient specific setting. Probably wouldn't be to hard to make a mechanism (akin to the user specific settings) that would support that.

**Comment by bradymiller** [comment]
looks good (just noted a couple lines that can be removed in config/config.yaml ). Looking forward to seeing files being removed (patientportal-style.css, etc. ) :)

File: library/globals.inc.php (hunk 2 above)
Comment (sjpadgett) [reply library/globals.inc.php:3068]: yep I'll persist somewhere in portal prob on new landing page. The global is a default.

**Comment by sjpadgett** [comment]
Removed documents from top level added to quickstart.

image

Now return to dashboard to exit.

image

I think I'm on the right track

**Comment by sjpadgett** [comment]
@bradymiller I've been thinking about dup email for register and think the security concerns outweigh the convivence.
I think Paul had a thought to use some sort of group or even guardian to put families together in the portal.

I'm still looking to find somewhere to store patient selected theme from portal. I could start a new table for portal account setup or I may just add to  patient_access_onsite

**Comment by sjpadgett** [comment]
I'll merge and pick up on reorg save and submit action buttons to be better available in document edit workflow. Will trim nav bar.
Then add module hook with example module.
Hope when bring in folks will look over on demo.

**Comment by bradymiller** [comment]
for the patient portal settings, i would rec a new table called `patient_settings` that looks basically the same as `user_settings` where using setting_pid instead if setting_user. This table can store globals and other non-global things (such as sticky settings like what happens when patient summary screen remembers to keeps certain widgets open).

**Comment by bradymiller** [comment]
2DV-2(12)

**Comment by sjpadgett** [comment]
I really like this suggestion. I can see it extremely useful throughout core.

**Comment by juggernautsei** [comment]
@sjpadgett is the portal backward compatible? If so, how can it be back ported?

**Comment by sjpadgett** [comment]
That is my hope Sherwin. Were you able to look at

**Comment by juggernautsei** [comment]
I just pulled it down this morning. I am going to bring it into a couple of my v7.0.0 installs to see how it runs.

**Comment by sjpadgett** [comment]
I'll see if I can put together a testing patch in next hour or two. Don't forget will have to run sql patch and I won't worry with the core theme removals.

**Comment by sjpadgett** [comment]
Also you should be able to look at on our dev Demo

**Comment by juggernautsei** [comment]
I brought the code into v7 (1) and these are the two errors so far. 
Solved:
     PHP Fatal error:  Uncaught Error: Class 'OpenEMR\\Services\\LogoService' not found in /var/www/html/openemr/portal/index.php:52\nStack trace:\n#0 {main}\n  thrown in /var/www/html/openemr/portal/index.php on line 52

Unsolved:
      PHP Fatal error:  Uncaught Error: Class 'Symfony\\Component\\Finder\\Finder' not found in /var/www/html/openemr/src/Services/LogoService.php:46\nStack trace:\n#0 /var/www/html/openemr/src/Services/LogoService.php(40): OpenEMR\\Services\\LogoService->resetFinder()\n#1 /var/www/html/openemr/portal/index.php(52): OpenEMR\\Services\\LogoService->__construct()\n#2 {main}\n  thrown in /var/www/html/openemr/src/Services/LogoService.php on line 46

I found this https://packagist.org/packages/symfony/finder. I am going to wait until after the backup of the server tonight before running the command to bring in the finder.

**Comment by sjpadgett** [comment]
That's because v7 production is missing these items and will need to be included in patch from master.
Robert is still working logo service and I'll need to find the best commits to bring back to v7 so give me the day while I test changes for next patch.

**Pull request merged**